pints/_mcmc/_monomial_gamma_hmc.py

```diff
@@ -0,0 +1,507 @@
+
+    In particular, the algorithm we implement follows eqs. (4.14)-(4.16) in
+    [2], since we allow different epsilon according to dimension.
+
+    [1] Towards Unifying Hamiltonian Monte Carlo and Slice Sampling
+    Yizhe Zhang, Xiangyu Wang, Changyou Chen, Ricardo Henao, Kai Fan,
+    Lawrence Cari.
+    [2] MCMC using Hamiltonian dynamics
+    Radford M. Neal, Chapter 5 of the Handbook of Markov Chain Monte
+    Carlo by Steve Brooks, Andrew Gelman, Galin Jones, and Xiao-Li Meng.
+
+    """
+    def __init__(self, x0, sigma0=None):
+        super(MonomialGammaHMCMCMC, self).__init__(x0, sigma0)
+
+        # Set initial state
+        self._running = False
+        self._ready_for_tell = False
+
+        # Current point in the Markov chain
+        self._current = None            # Aka current_q in the chapter
+        self._current_energy = None     # Aka U(current_q) = -log_pdf
+        self._current_gradient = None
+        self._current_momentum = None   # Aka current_p
+
+        # Current point in the leapfrog iterations
+        self._momentum = None       # Aka p in the chapter
+        self._position = None       # Aka q in the chapter
+        self._gradient = None       # Aka grad_U(q) in the chapter
+
+        # Iterations, acceptance monitoring, and leapfrog iterations
+        self._mcmc_iteration = 0
+        self._mcmc_acceptance = 0
+        self._frog_iteration = 0
+
+        # Default number of leapfrog iterations
+        self._n_frog_iterations = 20
+
+        # Default integration step size for leapfrog algorithm
+        self._epsilon = 0.1
+        self._step_size = None
+        self.set_leapfrog_step_size(np.diag(self._sigma0))
+
+        # Divergence checking
+        # Create a vector of divergent iterations
+        self._divergent = np.asarray([], dtype='int')
+
+        # Default threshold for Hamiltonian divergences
+        # (currently set to match Stan)
+        self._hamiltonian_threshold = 10**3
+
+        # Estimate quantities needed for sampling from the ke distribution
+        self._a = 1.0
+        self._c = 0.2
+        self._m = 1.0
+        self._initialise_ke()
+
+    def a(self):
+        """
+        Returns `a` in kinetic energy function
+        """
+        return self._a
+
+    def c(self):
+        """
+        Returns `c` in kinetic energy function
+        """
+        return self._c
+
+    def m(self):
+        """
+        Returns `m` in kinetic energy function
+        """
+        return self._m
+
+    def set_a(self, a):
+        """
+        Sets `a` in kinetic energy function
+        """
+        if a <= 0:
+            raise ValueError("a must be positive")
+        self._a = a
+        self._initialise_ke()
+
+    def set_c(self, c):
+        """
+        Sets `c` in kinetic energy function
+        """
+        if c <= 0:
+            raise ValueError("c must be positive")
+        self._c = c
+        self._initialise_ke()
+
+    def set_m(self, m):
```

Comment:
If its got a clear meaning, please use the conceptual name i.e. `set_mass` instead of a mathematical variable name